include/pybind11/cast.h

```diff
@@ -208,7 +208,7 @@ struct type_caster<T, enable_if_t<std::is_arithmetic<T>::value && !is_std_char_t
         return PyLong_FromUnsignedLongLong((unsigned long long) src);
     }
 
-    PYBIND11_TYPE_CASTER(T, _<std::is_integral<T>::value>("int", "float"));
+    PYBIND11_TYPE_CASTER(T, const_str<std::is_integral<T>::value>("int", "float"));
```

Comment:
I think we should add one test that keeps the `_` that only gets built/tested when `_` is not externally defined (so that we still test backwards compatibility for existing code still using `_`).